interfaces/builtin/lxd-support: allow LXD to manage its own cgroup

bboozzoo opened the pull request:
Set controls-device-cgroup flag for lxd-support interface, thus allowing LXD to self manage the device cgroup and also be exempt from the mandatory device cgroup set up with core24 base.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

mihalicyn commented:
Huge thanks, Maciej!

tomponline reviewed (commented):
Many thanks for the quick turn around on this!

alexmurray commented:
Similar to when we introduced the `enable-unconfined-mode` flag for the `lxd-support` interface, does this need to be behind a feature flag that the lxd snap can opt in to use? ie. Does a given lxd snap need to behave differently than it currently does when a snapd is in use that has this change? If so it feels like it would also need an interface attribute to be used to specify this.

bboozzoo commented:
> Similar to when we introduced the `enable-unconfined-mode` flag for the `lxd-support` interface, does this need to be behind a feature flag that the lxd snap can opt in to use? ie. Does a given lxd snap need to behave differently than it currently does when a snapd is in use that has this change? If so it feels like it would also need an interface attribute to be used to specify this.

Not really, it's a bug. The flag has been around since late 2020, introduced in https://github.com/snapcore/snapd/commit/35943584e567e74ca2ba8d372027ac24fb1a7523 for the exact purpose of skipping device cgroup setup for a given snap. This wasn't relevant to LXD, since the lxd interfaces are already powerful enough such that the snap does not have to request device access, but the bug surfaced only after the LXD team tried to switch to core24 which now uses a mandatory device cgroup. The necessary support for allowing interfaces to be explicitly exempt from device cgroup was added in https://github.com/snapcore/snapd/commit/69528a936a37ebd96f78366e3095e8b377d821ea because microovn broke (they plugged microcloud-support), As such docker-support, greengrass and microcloud-support interfaces have set the relevant flag, but I never checked lxd-support back when I proposed the branch.
